test/jdk/java/lang/invoke/defineHiddenClass/src/HiddenInterface.java

```diff
@@ -28,5 +28,6 @@
  */
 public interface HiddenInterface {
     default void test() {
+        record R() {}
```

Comment:
surprised to see this change

Reply:
this is was done because test: `test/jdk/java/lang/invoke/defineHiddenClass/PreviewHiddenClass.java` is compiling this code with options `--enable-preview -source [latestSource] -XDforcePreview` and then checking that the obtained class file has 65535 as its minor version. This was working so far as `DEFAULT_METHODS` was one member of `com.sun.tools.javac.code.Source.Feature` but as this PR is removing `DEFAULT_METHODS` we are obtaining the same effect defining a record as `RECORDS` are still defined in `com.sun.tools.javac.code.Source.Feature`

Reply:
yes that would be a sensible change going forward. Thanks for the review